### PR TITLE
Accept user-provided --proj and --tag

### PR DESCRIPTION
**Pull request by atheurer**
-defaults of localhost/workshop and latest should allow backward compat

-Also use rm and cp instead of mv, which appears to
 silently fail sometimes

### CODE REVIEW

File: workshop.pl [hunk 1]
```diff
@@ -1078,7 +1102,8 @@ sub arg_handler {
 
 if (-e $container_mount_point . "/etc/resolv.conf") {
     logger('info', "Backing up the temporary container's /etc/resolv.conf...\n");
-    ($command, $command_output, $rc) = run_command("mv --verbose " . $container_mount_point . "/etc/resolv.conf " . $container_mount_point . "/etc/resolv.conf." . $me);
+    ($command, $command_output, $rc) = run_command("/bin/cp --verbose --force " . $container_mount_point . "/etc/resolv.conf " . $container_mount_point . "/etc/resolv.conf.workshop");
+    ($command, $command_output, $rc) = run_command("/bin/rm --verbose --force " . $container_mount_point . "/etc/resolv.conf");
```
Comment (k-rister) [inline workshop.pl:1106]: Did making this change ultimately resolve whatever the problem you were seeing was?

File: workshop.pl [hunk 2]
```diff
@@ -1090,7 +1115,7 @@ sub arg_handler {
 }
 
 logger('info', "Temporarily copying the host's /etc/resolv.conf to the temporary container...\n");
-($command, $command_output, $rc) = run_command("cp --verbose /etc/resolv.conf " . $container_mount_point . "/etc/resolv.conf");
+($command, $command_output, $rc) = run_command("/bin/cp --verbose /etc/resolv.conf " . $container_mount_point . "/etc/resolv.conf");
```
Comment (k-rister) [inline workshop.pl:1118]: and this?

File: workshop.pl [hunk 3]
```diff
@@ -1493,7 +1518,7 @@ sub arg_handler {
 }
 
 logger('info', "Removing the temporarily assigned /etc/resolv.conf from the temporary container...\n");
-($command, $command_output, $rc) = run_command("rm --verbose " . $container_mount_point . "/etc/resolv.conf");
+($command, $command_output, $rc) = run_command("/bin/rm --verbose --force " . $container_mount_point . "/etc/resolv.conf");
```
Comment (k-rister) [inline workshop.pl:1521]: and this?

File: workshop.pl [hunk 4]
```diff
@@ -1503,9 +1528,9 @@ sub arg_handler {
 logger('info', "succeeded\n", 1);
 command_logger('verbose', $command, $rc, $command_output);
 
-if (-e $container_mount_point . "/etc/resolv.conf." . $me) {
+if (-e $container_mount_point . "/etc/resolv.conf.workshop") {
     logger('info', "Restoring the backup of the temporary container's /etc/resolv.conf...\n");
-    ($command, $command_output, $rc) = run_command("mv --verbose " . $container_mount_point . "/etc/resolv.conf." . $me . " " . $container_mount_point . "/etc/resolv.conf");
+    ($command, $command_output, $rc) = run_command("/bin/cp --verbose --force " . $container_mount_point . "/etc/resolv.conf.workshop " . $container_mount_point . "/etc/resolv.conf");
```
Comment (k-rister) [inline workshop.pl:1533]: and this?

**Review by k-rister — COMMENTED** [review]
I think I'm fine with these changes...but I would like a short pause to understand which, if any, were necessaryt to resolve the resolv.conf problem you were seeing versus which are just part of this change.

File: workshop.pl (hunk 4 above)
Comment (atheurer) [reply workshop.pl:1533]: I believe it is just the 'mv' that was problematic, and for some reason rm + cp instead seems to work more more reliably.  The full path on all of these is probably not necessary, just an extra precaution.  --force is also probably not necessary.

Comment (k-rister) [reply workshop.pl:1533]: Its not really important to this PR, but I sure wish I understand why you are getting better behavior from `rm+cp` compared to `mv`.

**Review by k-rister — APPROVED** [review]

**Pull request merged**